aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3Decorator.kt

```diff
@@ -79,6 +85,24 @@ class S3Decorator : ClientCodegenDecorator {
         it + S3PubUse()
     }
 
+    override fun endpointCustomizations(codegenContext: ClientCodegenContext): List<EndpointCustomization> {
+        return listOf(object : EndpointCustomization {
+            override fun setBuiltInOnServiceConfig(name: String, value: Node, configBuilderRef: String): Writable? {
+                if (name.startsWith("S3")) {
+                    return null
+                }
```

Comment:
What kinds of things does this skip? From my naïve point of view, it looks like we're filtering out stuff related to S3 in an S3 decorator, which I find strange.